bin/tape

```diff
@@ -41,18 +45,42 @@ if (typeof opts.ignore === 'string') {
     var matcher = ignore.createMatcher(ignoreStr);
 }
 
-opts._.forEach(function (arg) {
+var files = opts._.reduce(function (result, arg) {
     // If glob does not match, `files` will be an empty array.
     // Note: `glob.sync` may throw an error and crash the node process.
     var files = glob.sync(arg);
 
     if (!Array.isArray(files)) {
-        throw new TypeError('unknown error: glob.sync did not return an array or throw. Please report this.');
+        throw new TypeError('unknown error: glob.sync("' + arg + '") did not return an array or throw. Please report this.');
     }
 
-    files.filter(function (file) { return !matcher || !matcher.shouldIgnore(file); }).forEach(function (file) {
-        require(resolvePath(cwd, file));
-    });
+    return result.concat(files);
+}, [])
+    .filter(function (file) { return !matcher || !matcher.shouldIgnore(file); })
+    .map(function (file) { return resolvePath(cwd, file); });
+
+hasDynamicImport().then(function (hasSupport) {
+    var importOrRequire = hasSupport ? require('./import-or-require') : function (file) {
+        require(file);
+    };
+
+    var loadingPromise = files.reduce(function (loadingPromise, file) {
+        return loadingPromise ? loadingPromise.then(function () {
+            return importOrRequire(file);
+        }) : importOrRequire(file);
+    }, undefined);
+
+    if (loadingPromise) {
+        loadingPromise.then(
+          tape.run,
+          function (err) {
+            console.error(err);
+            process.exit(1);
```

Comment:
that unfortunately makes it more confusing, as the first output becomes

```
/home/notme/dev/tape/bin/tape:79
                setTimeout(function () { throw err; }, 0);
```

instead of saying where the error was originally thrown.